src/main/java/rcms/utilities/daqexpert/reasoning/logic/failures/deadtime/FedGeneratesDeadtime.java

```diff
@@ -0,0 +1,58 @@
+package rcms.utilities.daqexpert.reasoning.logic.failures.deadtime;
+
+import rcms.utilities.daqaggregator.data.DAQ;
+import rcms.utilities.daqaggregator.data.FED;
+import rcms.utilities.daqexpert.FailFastParameterReader;
+import rcms.utilities.daqexpert.Setting;
+import rcms.utilities.daqexpert.reasoning.logic.basic.FEDDeadtime;
+import rcms.utilities.daqexpert.reasoning.logic.basic.Parameterizable;
+import rcms.utilities.daqexpert.reasoning.logic.failures.KnownFailure;
+
+import java.util.Map;
+import java.util.Properties;
+
+public class FedGeneratesDeadtime extends KnownFailure implements Parameterizable {
+
+    private float deadtimeThresholdInPercentage;
+    private float backpressureThresholdInPercentage;
+
+    public FedGeneratesDeadtime() {
+        this.name = "FED problem";
+    }
+
+    @Override
+    public boolean satisfied(DAQ daq, Map<String, Boolean> results) {
+
+        boolean fedDeadtime = results.get(FEDDeadtime.class.getSimpleName());
+        if (!fedDeadtime) {
+            return false;
+        }
+        boolean result = false;
+
+
+        for (FED fed : daq.getFeds()) {
+            float deadPercentage = 0;
+            float backpressure = fed.getPercentBackpressure();
+            deadPercentage += fed.getPercentBusy();
+            deadPercentage += fed.getPercentWarning();
+
+            if (deadPercentage > deadtimeThresholdInPercentage && backpressure < backpressureThresholdInPercentage) {
+                result = true;
+                context.register("FED",fed.getSrcIdExpected());
+                context.registerForStatistics("DEADTIME",deadPercentage,"%",1);
+                context.registerForStatistics("BACKPRESSURE",deadPercentage,"%",1);
+
```

Comment:
yes, this looks like what is described in #155 . 

The statistics variables will capture the mean, min and max not only over time but also over FEDs in case there is more than one FED backpressured by the DAQ (which is ok in my opinion).

Reply:
I agree, otherwise reporting the statistics per fed over time would be very verbose.